lib/mayaUsd/ufe/StagesSubject.cpp

```diff
@@ -298,29 +321,42 @@ void StagesSubject::stageChanged(
             // the add or delete of our UFE/USD implementation.
             if (InAddOrDeleteOperation::inAddOrDeleteOperation()) {
                 if (prim.IsActive()) {
-#ifdef UFE_V2_FEATURES_AVAILABLE
-                    Ufe::Scene::instance().notify(Ufe::ObjectAdd(sceneItem));
-#else
-                    auto notification = Ufe::ObjectAdd(sceneItem);
-                    Ufe::Scene::notifyObjectAdd(notification);
-#endif
+                    sendObjectAdd(sceneItem);
                 } else {
+                    sendObjectPostDelete(sceneItem);
+                }
+            } else {
+                // Use the entry flags in the USD notice to know what operation was performed and
+                // thus what Ufe notif to send.
+                const std::vector<const SdfChangeList::Entry*>& entries = it.base()->second;
+                bool                                            sentNotif { false };
+                for (const auto& entry : entries) {
+                    if (entry->flags.didAddInertPrim || entry->flags.didAddNonInertPrim) {
+                        sendObjectAdd(sceneItem);
+                        sentNotif = true;
+                        break;
+                    } else if (
+                        entry->flags.didRemoveInertPrim || entry->flags.didRemoveNonInertPrim) {
+                        sendObjectPostDelete(sceneItem);
+                        sentNotif = true;
+                        break;
+                    }
+                }
+
+                if (!sentNotif) {
 #ifdef UFE_V2_FEATURES_AVAILABLE
-                    Ufe::Scene::instance().notify(Ufe::ObjectPostDelete(sceneItem));
+                    // According to USD docs for GetResyncedPaths():
+                    // - Resyncs imply entire subtree invalidation of all descendant prims and
+                    // properties. So we send the UFE subtree invalidate notif.
+                    Ufe::Scene::instance().notify(Ufe::SubtreeInvalidate(sceneItem));
```

Comment:
If we didn't find a more appropriate notif, we still send the subtree invalidate (as before). But we also mimic that below (with add/delete) for Ufe v1.